iothub/device/samples/getting started/SimulatedDeviceWithCommand/SimulatedDeviceWithCommand.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    internal class SimulatedDeviceWithCommand
+    {
+        private readonly TimeSpan? _maxRunTime;
+        private readonly IotHubDeviceClient _deviceClient;
+        private static TimeSpan s_telemetryInterval;
+
+        public SimulatedDeviceWithCommand(IotHubDeviceClient deviceClient, TimeSpan? maxRunTime)
+        {
+            _deviceClient = deviceClient ?? throw new ArgumentNullException(nameof(deviceClient));
+            _maxRunTime = maxRunTime;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            using CancellationTokenSource cts = _maxRunTime.HasValue
+                ? new CancellationTokenSource(_maxRunTime.Value)
+                : new CancellationTokenSource();
+            Console.CancelKeyPress += (sender, eventArgs) =>
+            {
+                eventArgs.Cancel = true;
+                cts.Cancel();
+                Console.WriteLine("Sample execution cancellation requested; will exit.");
+            };
+
+            Console.WriteLine($"{DateTime.Now}> Press Control+C at any time to quit the sample.");
+
+            try
+            {
+                await _deviceClient.OpenAsync(cts.Token);
+
+                await _deviceClient.SetDirectMethodCallbackAsync(SetTelemetryInterval, cts.Token);
+
+                await SendDeviceToCloudMessagesAsync(cts.Token);
+            }
+            catch (OperationCanceledException) { }
+        }
+
+        // Handle the direct method call.
+        private Task<DirectMethodResponse> SetTelemetryInterval(DirectMethodRequest methodRequest)
+        {
+            string data = methodRequest.GetPayloadAsJsonString();
```

Comment:
Why not `methodRequest.TryGetPayload<int>(out int telemetryIntervalSeconds);`?

Reply:
Updated